client.go

```diff
@@ -113,19 +116,21 @@ func (r *Resolver) Lookup(ctx context.Context, instance, service, domain string,
 		params.Domain = domain
 	}
 	params.Entries = entries
-
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
I would argue that this type of ctx use is still a misuse in the Golang sense. 
Top-level context should be always the caller (browse etc.). What we're (I did already before) here is to branch off and reuse the internal channel of the current context as signaling mechanism.
Not sure whether we should avoid this in total.

Reply:
Well what you argue is a best practice and doesn't make this change a misuse. The issue was that cancel didn't work at all before this change(i.e. you didn't cancel anything). Basically this change fixed a bug.  I believe refactoring/best practices should be a different issue.

Reply:
Thanks for clarifying. I have not been aware that it is totally broken.

I'll check that issue ober the weekend ans then merge it.

Reply:
Below is a simple snippet proving that line 97 ( https://github.com/grandcat/zeroconf/pull/30/files/43e06c6664e3398ec654c65452b94f392670c757#diff-d0b7f86cea13f90d3cc0cc2338da8d5fL97) simply didn't work as the parent context was not cancelled. WithCancel cancels only the descendents not the ancestors.

https://play.golang.org/p/aFhmy4hqe7